include/swift/AST/DiagnosticsSema.def

```diff
@@ -6756,6 +6756,15 @@ ERROR(availability_parameterized_protocol_only_version_newer, none,
       "runtime support for parameterized protocol types is only available in "
       "%0 %1 or newer",
       (StringRef, llvm::VersionTuple))
+      
+ERROR(non_runtime_objc_protocol_metadata_not_available,none,
+      "this Objective-C protocol '%0' has the non_runtime_protocol attribute"
+      " and its metadata is not available in Swift.",
+      (StringRef))
+
+NOTE(non_runtime_objc_protocol_metadata_not_available_note,none,
```

Comment:
Just checked , the `decl_declared_here` family are all using swift types, e.g. ` ValueDecl` `DeclName`,  https://github.com/swiftlang/swift/blob/main/include/swift/AST/DiagnosticsSema.def#L30

not clang decl types, like `const clang::NamedDecl*`, do you suggest I add a new `decl_declared_here` with the clang type, or just keep this `non_runtime_objc_protocol_metadata_not_available_note`?

Reply:
Yes, I think it would be worth introducing a generic diagnostic string that's compatible with this use case, rather than creating a diagnostic with a very specific name but very generic text.